code/datums/armor/_armor.dm

```diff
@@ -0,0 +1,221 @@
+		.[armor_type.type] = armor_type
+		armor_type.GenerateTag()
+
+/proc/get_armor_by_type(armor_type)
+	. = locate(replacetext("[armor_type]", "/", "-"))
+	if(.)
+		return .
+	if(armor_type == /datum/armor)
+		CRASH("Attempted to get the base armor type, you probably meant to use /datum/armor/none")
+	CRASH("Attempted to get an armor type that did not exist! '[armor_type]'")
+
+/**
+ * The armor datum holds information about different types of armor that an atom can have.
+ * It also contains logic and helpers for calculating damage and effective damage
+ */
+/datum/armor
+	VAR_PROTECTED/acid = 0
+	VAR_PROTECTED/bio = 0
+	VAR_PROTECTED/bomb = 0
+	VAR_PROTECTED/bullet = 0
+	VAR_PROTECTED/consume = 0
+	VAR_PROTECTED/energy = 0
+	VAR_PROTECTED/laser = 0
+	VAR_PROTECTED/fire = 0
+	VAR_PROTECTED/melee = 0
+	VAR_PROTECTED/wound = 0
+
+/// A version of armor with no protections
+/datum/armor/none
+
+/// A version of armor that cannot be modified and will always return itself when attempted to be modified
+/datum/armor/immune
+
+/datum/armor/Destroy(force, ...)
+	if(!force && tag)
+		return QDEL_HINT_LETMELIVE
+
+	// something really wants us gone
+	datum_flags &= ~DF_USE_TAG
+	tag = null
+	return ..()
+
+/datum/armor/GenerateTag()
+	..()
+	tag = replacetext("[type]", "/", "-")
+
+/datum/armor/vv_edit_var(var_name, var_value)
+	return FALSE
+
+/datum/armor/can_vv_mark()
+	return FALSE
+
+/datum/armor/vv_get_dropdown()
+	return list("", "MUST MODIFY ARMOR VALUES ON THE PARENT ATOM")
+
+/datum/armor/CanProcCall(procname)
+	return FALSE
+
+/// Generate a brand new armor datum with the modifiers given, if ARMOR_ALL is specified only that modifier is used
+/datum/armor/proc/generate_new_with_modifiers(list/modifiers)
+	var/datum/armor/new_armor = new
+
+	var/all_keys = ARMOR_LIST_ALL()
+	if(ARMOR_ALL in modifiers)
+		var/mod_all = modifiers[ARMOR_ALL]
+		if(!mod_all)
+			return src
+		for(var/mod in all_keys)
+			new_armor.vars[mod] = vars[mod] + mod_all
+		return new_armor
+
+	for(var/mod in modifiers)
+		if(!(mod in all_keys)) // Did you know that the false block evaluates faster than the true block?
+			stack_trace("Attempt to call generate_new_with_modifiers with illegal modifier '[mod]'! ignoring it")
+		else
+			new_armor.vars[mod] = vars[mod] + modifiers[mod]
+	return new_armor
+
+/datum/armor/immune/generate_new_with_modifiers(list/modifiers)
+	return src
+
+/// Generate a brand new armor datum with the multiplier given, if ARMOR_ALL is specified only that modifer is used
+/datum/armor/proc/generate_new_with_multipliers(list/multipliers)
+	var/datum/armor/new_armor = new
+
+	var/all_keys = ARMOR_LIST_ALL()
+	if(ARMOR_ALL in multipliers)
+		var/mult_all = multipliers[ARMOR_ALL]
+		if(!mult_all)
+			return src
+		for(var/mod in all_keys)
+			new_armor.vars[mod] = vars[mod] * mult_all
+		return new_armor
+
+	for(var/mod in multipliers)
+		if(!(mod in all_keys)) // Did you know that the false block evaluates faster than the true block?
```

Comment:
Invert it, `if(!x) else` can be hard to interpret and the speed buff here is barely even noticeable in space init, much less here